mlflow/metrics/__init__.py

```diff
@@ -35,333 +35,365 @@
+        version="v1",
+    )
+
+
+@experimental
+def flesch_kincaid_grade_level() -> EvaluationMetric:
+    """
+    This function will create a metric for calculating `flesch kincaid grade level`_ using
+    `textstat`_.
+
+    This metric outputs a number that approximates the grade level needed to comprehend the text,
+    which will likely range from around 0 to 15 (although it is not limited to this range).
+
+    Aggregations calculated for this metric:
+        - mean
+
+    .. _flesch kincaid grade level:
+        https://en.wikipedia.org/wiki/Flesch%E2%80%93Kincaid_readability_tests#Flesch%E2%80%93Kincaid_grade_level
+    .. _textstat: https://pypi.org/project/textstat/
+    """
+    return make_metric(
+        eval_fn=_flesch_kincaid_eval_fn,
+        greater_is_better=False,
+        name="flesch_kincaid_grade_level",
+        version="v1",
+    )
+
+
+@experimental
+def ari_grade_level() -> EvaluationMetric:
+    """
+    This function will create a metric for calculating `automated readability index`_ using
+    `textstat`_.
+
+    This metric outputs a number that approximates the grade level needed to comprehend the text,
+    which will likely range from around 0 to 15 (although it is not limited to this range).
+
+    Aggregations calculated for this metric:
+        - mean
+
+    .. _automated readability index: https://en.wikipedia.org/wiki/Automated_readability_index
+    .. _textstat: https://pypi.org/project/textstat/
+    """
+    return make_metric(
+        eval_fn=_ari_eval_fn,
+        greater_is_better=False,
+        name="ari_grade_level",
+        long_name="automated_readability_index_grade_level",
+        version="v1",
+    )
 
-.. _automated readability index: https://en.wikipedia.org/wiki/Automated_readability_index
-.. _textstat: https://pypi.org/project/textstat/
-"""
 
 # question answering metrics
+@experimental
+def exact_match() -> EvaluationMetric:
+    """
+    This function will create a metric for calculating `accuracy`_ using sklearn.
 
-exact_match = make_metric(
-    eval_fn=_accuracy_eval_fn, greater_is_better=True, name="exact_match", version="v1"
-)
-exact_match.__doc__ = """
-.. Note:: Experimental: This metric may change or be removed in a future release without warning.
+    This metric only computes an aggregate score which ranges from 0 to 1.
 
-A metric for calculating `accuracy`_ using sklearn.
+    .. _accuracy: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html
+    """
+    return make_metric(
+        eval_fn=_accuracy_eval_fn, greater_is_better=True, name="exact_match", version="v1"
+    )
 
-This metric only computes an aggregate score which ranges from 0 to 1.
-
-.. _accuracy: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html
-"""
 
 # text summarization metrics
+@experimental
+def rouge1() -> EvaluationMetric:
+    """
+    This function will create a metric for evaluating `rouge1`_.
```

Comment:
curious why do we have a `_` outside backtick?

Reply:
this is for hyperlinking